### ISSUE TITLE
MSVC build issue - Does not build on Japanese system locale without changing code

### ISSUE THREAD

**Issue by Dwedit** [libretro/RetroArch#6404]
## Description
MSVC has code page issues when compiling source files.  If your system locale is set to Windows-1252 (Western), there are no problems.  However, if your system locale is set to code page Windows-932 (Japanese: Shift-JIS), you can't build "msg_hash_vn.h" or "msg_hash_ja.h"  The UTF-8 bytes paired with ASCII characters get misidentified as Shift-JIS characters and cause errors.

The solution in MSVC is to use the /utf-8 switch when compiling, which forces it to ignore the system codepage, and treat the source and output files as UTF-8 encoded text.

However, for some whatever reason, #pragma execution_character_set("utf-8") cannot be used if you specified /utf-8 on the command line, even though it is changing the execution character set to the same thing.  You get a compiler error.

The /utf-8 switch was introduced in Visual Studio 2015.  The  #pragma execution_character_set feature was introduced in Visual Studio 2008, and was deprecated as of Visual Studio 2015.

So I propose changing this block:
```
#if defined(_MSC_VER) && !defined(_XBOX)
/* https://support.microsoft.com/en-us/kb/980263 */
#pragma execution_character_set("utf-8")
#endif
```

To this:
```
#if (_MSC_VER >= 1500 && _MSC_VER < 1900) && !defined(_XBOX)
/* https://support.microsoft.com/en-us/kb/980263 */
#pragma execution_character_set("utf-8")
#endif
```

Then define /utf-8 as a command line switch for the MSVC 2015 and 2017 projects.

This will make it continue to build on MSVC 2008-2013 compilers for systems using Western locale, then work on JP locale systems for MSVC 2015 and later.

### Environment information

- OS: Windows 10, with System Code Page set to 932
- Compiler: MSVC 2017

**Comment by bparker06** [comment]
Hi @Dwedit,

Currently what we do is to make sure files that contain utf-8 have their BOM set at the beginning of the file, this tells all utf8-aware versions of MSVC to set that as the encoding. However, people inadvertently remove these BOMs *all the time*, and those removals continue to be merged (it doesn't help that github doesn't show it was removed either), so it is something I have always had to deal with, especially for older MSVC versions (where there is no `/utf-8`). However adding that flag for later supported versions might work as a stop-gap for the times when the BOM is missing... my only real concern is `media/rarch_ja.rc` which has to stay in the CP932 encoding for legacy reasons and I'm not sure if `/utf-8` would mess with that.

I added the BOM back to the ar/vn files they were missing in and compilation seems to proceed now on a CP932 system.
